.circleci/config.yml

```diff
@@ -343,6 +343,86 @@ jobs:
                       # taken from slack-secrets context
                       channel: << parameters.notify_on_failure_channel >>
 
+  unit-tests-new:
+    parameters:
+      python_version_major: {type: integer}
+      python_version_minor: {type: integer}
+      with_core: {type: boolean}
+      tests:
+        type: enum
+        enum: [sdk, other]
+
+    docker:
+      - image: python:<<parameters.python_version_major>>.<<parameters.python_version_minor>>
+
+    # We're testing some heavy "units". Ideally we would use the default,
+    # but for that we must pare down the unit tests.
+    resource_class: xlarge
+
+    # Shards the job, setting $CIRCLE_NODE_TOTAL and $CIRCLE_NODE_INDEX.
+    # https://circleci.com/docs/parallelism-faster-jobs/
+    parallelism: 2
+
+    steps:
+      - checkout
+      - install_go
+
+      - run:
+          name: Install system deps
+          command: |
+            apt-get update
+            apt-get install -y libsndfile1 ffmpeg
+
+      - run:
+          name: Install Python dependencies
+          command: pip install -U tox nox pip
+          no_output_timeout: 5m
+
+      # For now, this step needs to be duplicated because there's no way
```

Comment:
you can control it with the tox env name... if you don't want to repeat yourself
basically have `core-py37-sdk`,  and `core-py37-other`

Reply:
just need to do a small update to the tox file...

Reply:
It would move the ugliness to tox.ini. Right now that section expects the paths to run to be passed via positional arguments, so if I don't want to refactor all other callsites, then I'd need to add {,-sdk,-other} factors, and then use three commands like this:

```
sdk: pytest ...
other: pytest ...
!sdk,!other: pytest ...
```

since the goal is to move tests to nox anyway, I was thinking of avoiding touching the tox config for now

Reply:
yeah, i agree with this one. just pointed out that this is an option, i don't mind the temporary conditional thing here... ideally we get away from folders and user markers and the markers can be mapped to the types here...